src/tweenable.rs

```diff
@@ -561,72 +551,122 @@ impl<T> Tweenable<T> for Sequence<T> {
+                } else {
+                    // We're right on the boundary of completing this tween, so mark it complete.
+                    if index >= self.index {
+                        tween.set_progress(1.);
+                    }
+                    index += 1;
+                }
+
+                break;
             }
-            tween.set_progress(1.); // ?? to prepare for next loop/rewind?
-            accum_duration += tween_duration;
+
+            index += 1;
+            delta -= tween_duration;
         }
 
-        // None found; sequence ended
-        self.index = self.tweens.len();
+        if index < self.index {
+            let end = min(self.index + 1, self.tweens.len());
+            for tween in &mut self.tweens[index + 1..end] {
+                tween.rewind();
+            }
+        }
+        self.index = index;
     }
 
     fn progress(&self) -> f32 {
-        self.time.as_secs_f32() / self.duration.as_secs_f32()
+        self.elapsed.as_secs_f32() / self.duration.as_secs_f32()
     }
 
     fn tick(
         &mut self,
-        delta: Duration,
+        mut delta: Duration,
         target: &mut T,
         entity: Entity,
         event_writer: &mut EventWriter<TweenCompleted>,
     ) -> TweenState {
-        if self.index < self.tweens.len() {
-            let mut state = TweenState::Active;
-            self.time = (self.time + delta).min(self.duration);
-            let tween = &mut self.tweens[self.index];
-            let tween_state = tween.tick(delta, target, entity, event_writer);
-            if tween_state == TweenState::Completed {
-                tween.rewind();
-                self.index += 1;
-                if self.index >= self.tweens.len() {
-                    state = TweenState::Completed;
-                    self.times_completed = 1;
-                }
+        self.elapsed = min(self.elapsed + delta, self.duration);
+
+        let len = self.tweens.len();
+        let mut state = TweenState::Completed;
+        for tween in &mut self.tweens[self.index..] {
+            let prev_progress = tween.progress();
+            let prev_completions = tween.times_completed();
+
+            state = tween.tick(delta, target, entity, event_writer);
+            if state != TweenState::Completed {
+                // If we completed zero times, then that means the entire delta was used up on this
+                // tween. Otherwise, we need to diff the tween progress because it overlaps the
+                // completion boundary.
+                break;
+            }
+            self.index += 1;
+            if self.index == len {
+                // We've reached the end so we don't care about the remaining delta.
+                break;
+            }
+
+            let tween_duration = tween.duration();
+
+            let full_completions =
+                (tween.times_completed() - prev_completions - 1) * tween_duration;
```

Comment:
Note that this seems to attempt to handle a case that is currently not supported, namely having looping tweenables as part of a `Sequence`. Currently `full_completions` will always be zero. Also, if a tweenable is looping, it currently never return `TweenState::Completed` since looping tweenables are of infinite duration by definition.